src/test/spec/unified_runner/operation.rs

```diff
@@ -1919,6 +1938,195 @@ impl TestOperation for RenameCollection {
+            if let Some(id) = &self.store_errors_as_entity {
+                let errors = Bson::Array(vec![]);
+                test_runner.insert_entity(id, errors.into());
+            }
+            if let Some(id) = &self.store_failures_as_entity {
+                let failures = Bson::Array(vec![]);
+                test_runner.insert_entity(id, failures.into());
+            }
+            if let Some(id) = &self.store_successes_as_entity {
+                let successes = Bson::Int64(0);
+                test_runner.insert_entity(id, successes.into());
+            }
+            if let Some(id) = &self.store_iterations_as_entity {
+                let iterations = Bson::Int64(0);
+                test_runner.insert_entity(id, iterations.into());
+            }
+
+            let continue_looping = Arc::new(AtomicBool::new(true));
+            let continue_looping_handle = continue_looping.clone();
+            ctrlc::set_handler(move || continue_looping_handle.store(false, Ordering::SeqCst))
+                .expect("Failed to set ctrl-c handler");
+
+            while continue_looping.load(Ordering::SeqCst) {
+                for operation in &self.operations {
+                    let result = match operation.object {
+                        OperationObject::TestRunner => {
+                            panic!("Operations within a loop must be entity operations")
+                        }
+                        OperationObject::Entity(ref id) => {
+                            operation.execute_entity_operation(id, test_runner).await
+                        }
+                    };
+                    match result {
+                        Ok(entity) => match operation.expectation {
+                            Expectation::Result {
+                                ref expected_value,
+                                ref save_as_entity,
+                            } => {
+                                if let Some(expected_value) = expected_value {
+                                    if let Some(Entity::Bson(ref actual_value)) = entity {
+                                        if actual_value == expected_value {
+                                            self.report_success(test_runner);
+                                        } else {
+                                            self.report_failure(
+                                                &operation.name,
+                                                actual_value,
+                                                expected_value,
+                                                test_runner,
+                                            );
+                                            // If an unexpected failure is encountered, the
+                                            // sub-operation loop should be terminated.
+                                            break;
+                                        }
+                                    }
```

Comment:
Should there be handling for expected value but the entity is `None` / not `Bson`?